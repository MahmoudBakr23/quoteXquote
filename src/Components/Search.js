import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ handleSearch }) => (
  <div className="search-select">
    <input type="text" name="search" onChange={(e) => handleSearch(e)} placeholder="search..." />
    <ul className="hint">
      <li><h6>Naruto</h6></li>
      <li><h6>Steins;Gate</h6></li>
      <li><h6>Kimetsu no Yaiba</h6></li>
      <li><h6>Akame ga Kill!</h6></li>
      <li><h6>Bleach</h6></li>
    </ul>
  </div>
);

export default Search;

Search.propTypes = {
  handleSearch: PropTypes.func,
};

Search.defaultProps = {
  handleSearch: null,
};
