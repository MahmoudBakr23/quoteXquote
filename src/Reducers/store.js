import { createStore } from 'redux';
import animeReducer from './anime';

const store = createStore(animeReducer);

export default store;
