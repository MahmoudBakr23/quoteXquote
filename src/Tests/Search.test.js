import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Reducers/store';
import Search from '../Components/Search';

describe('Test Anime List Component', () => {
  const anime = render(<Search />);

  test('Testing elements', () => {
    anime.getByText('Bleach');
    anime.getByText('Kimetsu no Yaiba');
    anime.getByText('Naruto');
  });

  it('Expect to render Search component', () => {
    const result = renderer.create(
      <Provider store={store}>
        <Search />
      </Provider>,
    ).toJSON;
    expect(result).toMatchSnapshot();
  });
});
