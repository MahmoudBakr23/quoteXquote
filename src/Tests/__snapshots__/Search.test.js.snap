// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Anime List Component Expect to render Search component 1`] = `[Function]`;
